src/Pay/Exception.php

```diff
@@ -16,10 +16,18 @@ class Exception extends \Exception
 
     protected string $type = '';
 
-    public function __construct(string $type = Exception::GENERAL_UNKNOWN, string $message = null, int $code = null, \Throwable $previous = null)
+    /**
+     * Error object with additional error data
+     *
+     * @var array
+     */
+    protected array $error = [];
```

Comment:
Let's name it `metadata`, the name `error` is a bit confusing inside an exception

Reply:
updated